package_control/package_manager.py

```diff
@@ -1600,6 +1601,43 @@ def install_package(self, package_name, unattended=False):
                     fail_early=False
                 )
 
+            try:
+                extra_pc_settings_json = package_zip.read(common_folder + 'Package Control.sublime-settings')
+                extra_pc_settings = json.loads(extra_pc_settings_json.decode('utf-8'))
+            except (KeyError):
+                pass
+            except (ValueError):
+                console_write(
+                    '''
+                    Failed to parse the Package Control.sublime-settings for "%s"
+                    ''',
+                    package_name
+                )
+            else:
+                wanted_packages = set(extra_pc_settings.get('installed_packages') or [])
+                pc_settings = sublime.load_settings(pc_settings_filename())
+                in_process_packages = load_list_setting(pc_settings, 'in_process_packages')
+                additional_packages = wanted_packages - in_process_packages - self.installed_packages()
+                if additional_packages:
+                    task_runner = PackageTaskRunner(self)
```

Comment:
Recursive task runners, are likely to bring back all the race conditions of PC3. The point of those runners is to carefully batch installs/updates so they can be performed in one go:

1. backup active theme/colorscheme/syntax assignments
2. disable all packages
3. install/update all packages (and install required libs)
4. re-enable all packages
5. restore theme/color scheme/syntax

This commit violates this strategy.